superset/databases/filters.py

```diff
@@ -41,6 +41,16 @@ class DatabaseFilter(BaseFilter):  # pylint: disable=too-few-public-methods
     # TODO(bogdan): consider caching.
 
     def apply(self, query: Query, value: Any) -> Query:
+        # Dynamic Filters need to be applied to the Query before we filter
+        # databases with anything else. This way you can show/hide databases using
+        # Feature Flags for example in conjuction with the regular role filtering.
+        # If not, if an user has access to all Databases it would skip this dynamic
+        # filtering.
+
+        if dynamic_filter := current_app.config["EXTRA_DYNAMIC_DATABASE_FILTER"]:
+            query = dynamic_filter(query)
```

Comment:
what happens when the `dynamic_filter` isn't defined? looking at the code it seems like ` return query.filter(` will fail.

Can we add a condition to check if the `dynamic_filter` is defined, then have a test to back it

Reply:
Hmm how so? `dynamic_filter` is only defined if the `EXTRA_DYNAMIC_DATABASE_FILTER ` is defined in the config and if so, it will apply it to the query, otherwise the code is never going to enter into this conditional.